packages/react-aria-components/test/AriaAutocomplete.test-util.tsx

```diff
@@ -0,0 +1,561 @@
+          let {getByRole} = renderers.standard({autocompleteProps: {defaultInputValue: 'Ba'}});
+          let input = getByRole('searchbox');
+          expect(input).toHaveValue('Ba');
+          let menu = getByRole(collectionNodeRole);
+          let options = within(menu).getAllByRole(collectionItemRole);
+          expect(options).toHaveLength(2);
+          expect(options[0]).toHaveTextContent('Bar');
+          expect(options[1]).toHaveTextContent('Baz');
+
+          await user.tab();
+          expect(document.activeElement).toBe(input);
+          await user.keyboard('z');
+          act(() => jest.runAllTimers());
+          options = within(menu).getAllByRole(collectionItemRole);
+          expect(options).toHaveLength(1);
+          expect(options[0]).toHaveTextContent('Baz');
+        });
+
+        it('should support keyboard navigation', async function () {
+          let {getByRole} = renderers.standard({});
+          let input = getByRole('searchbox');
+          let menu = getByRole(collectionNodeRole);
+          let options = within(menu).getAllByRole(collectionItemRole);
+          expect(input).not.toHaveAttribute('aria-activedescendant');
+
+          await user.tab();
+          expect(document.activeElement).toBe(input);
+
+          await user.keyboard('{ArrowDown}');
+          expect(input).toHaveAttribute('aria-activedescendant', options[0].id);
+          await user.keyboard('{ArrowDown}');
+          expect(input).toHaveAttribute('aria-activedescendant', options[1].id);
+          await user.keyboard('{ArrowDown}');
+          expect(input).toHaveAttribute('aria-activedescendant', options[2].id);
+          await user.keyboard('{ArrowUp}');
+          expect(input).toHaveAttribute('aria-activedescendant', options[1].id);
+
+          expect(document.activeElement).toBe(input);
+        });
+
+        it('should clear the focused key when using ArrowLeft and ArrowRight', async function () {
+          let {getByRole} = renderers.standard({});
+          let input = getByRole('searchbox');
+          let menu = getByRole(collectionNodeRole);
+          expect(input).not.toHaveAttribute('aria-activedescendant');
+
+          await user.tab();
+          expect(document.activeElement).toBe(input);
+
+          await user.keyboard('Foo');
+          act(() => jest.runAllTimers());
+          let options = within(menu).getAllByRole(collectionItemRole);
+          expect(input).toHaveAttribute('aria-activedescendant', options[0].id);
+          await user.keyboard('{ArrowRight}');
+          expect(input).not.toHaveAttribute('aria-activedescendant');
+          await user.keyboard('{ArrowDown}');
+          expect(input).toHaveAttribute('aria-activedescendant', options[0].id);
+          await user.keyboard('{ArrowLeft}');
+          expect(input).not.toHaveAttribute('aria-activedescendant');
+          expect(document.activeElement).toBe(input);
+        });
+
+        if (collectionType === 'menu') {
```

Comment:
i do not love splitting this into more 'if's, I was hoping that the top level would only be renderer `if`'s

at the very least, i'd like for it not to be collection related, could we change it to `role`? or ariaPattern?

Reply:
I'll change it to `ariaPattern`. Its a bit unfortunate that I had to split it out like this but I wanted to test the event forwarding to the wrapped components in a manner where I didn't have to duplicate the tests per each combination